crates/uv-cli/src/lib.rs

```diff
@@ -4797,6 +4797,22 @@ pub struct PublishArgs {
         value_parser = parse_insecure_host,
     )]
     pub allow_insecure_host: Option<Vec<Maybe<TrustedHost>>>,
+
+    /// Don't upload files that already exists on the index. The value is the index URL.
+    ///
+    /// This option allows retrying publishing that failed after only some, but not all files have
+    /// been uploaded, and handles error due to parallel uploads of the same file.
+    ///
+    /// Before uploading, the index is checked. If the exact same file already exists in the index,
+    /// the file will not be uploaded. If an error occurred during the upload, the index is checked
+    /// again, if the file may have been added in the meantime.
+    ///
+    /// The exact behavior varies from index to index. On PyPI, uploading the same file succeeds
+    /// even without `--skip-existing`, while most other indexes error.
+    ///
+    /// The index must provide one of the supported hashes (SHA-256, SHA-384, or SHA-512).
+    #[arg(long,env = EnvVars::UV_PUBLISH_SKIP_EXISTING)]
+    pub skip_existing: Option<IndexUrl>,
```

Comment:
Why does this take an index, rather than a bool?

Reply:
But why do we need to provide the index here? Because you need to be able to query for metadata, and the publish URL isn't sufficient?

Reply:
I'm not overly attached to doing it this way, i mainly picked it because it seemed like a clear, explainable design.

Reply:
As I've said above, the name is open for bike shedding. I like it for being self explaining (`uv publish --skip-existing https://example.org/simple` means to skip any files existing on https://example.org/simple when publishing) and for not being confused with the regular index API.